ui/prometheus-plugin/src/utils/utils.test.ts

```diff
@@ -160,10 +160,18 @@ describe('getFormattedPrometheusSeriesName', () => {
   it('should resolve empty metric to instead show query', () => {
     const query = 'node_load15{instance=~"(demo.do.prometheus.io:9100)"';
     const metric = {};
-    const output = { name: query };
+    const output = { name: query, formattedName: query };
     expect(getFormattedPrometheusSeriesName(query, metric)).toEqual(output);
   });
 
+  it('should resolve empty metric with formatted series name', () => {
+    const query = 'up';
+    const metric = {};
+    const series_name_format = 'Custom series name';
+    const output = { name: query, formattedName: 'Custom series name' };
```

Comment:
I wonder if it's worth breaking out just the basics of that panel story into a separate PR from the query refactor, so we can get it out for use faster. No strong feels about it, but might be something you could collaborate with Eunice on since it seems related to a couple of your PRs.

Reply:
I did end up adding a panel story in a different branch, I think this bug has decent coverage from the unit tests now so going to go ahead and merge. Planning to add visual regression tests for more of the tooltip and legend behavior later